src/System.Web.OData/OData/Builder/StructuralTypeConfiguration.cs

```diff
@@ -211,9 +211,27 @@ public virtual IEnumerable<NavigationPropertyConfiguration> NavigationProperties
         protected internal IList<PropertyInfo> RemovedProperties { get; private set; }
 
         /// <summary>
+        /// Variable for property ExplicitProperties.
+        /// </summary>
+        protected IDictionary<PropertyInfo, PropertyConfiguration> explicitInnerProperties;
+
+        /// <summary>
         /// Gets the collection of explicitly added properties.
         /// </summary>
-        protected internal IDictionary<PropertyInfo, PropertyConfiguration> ExplicitProperties { get; private set; }
+        protected internal IDictionary<PropertyInfo, PropertyConfiguration> ExplicitProperties
+        {
+            get
+            {
+                if (explicitInnerProperties.Any(a => a.Value.QueryConfiguration != null &&
+                        a.Value.QueryConfiguration.ModelBoundQuerySettings != null))
+                {
+                    explicitInnerProperties = OrderColumn(explicitInnerProperties);
+                }
+
+                return explicitInnerProperties;
+            }
+            private set => explicitInnerProperties = value;
```

Comment:
change to this?
private set { explicitInnerProperties = value; }

Reply:
Shall we set C# version in csproj, instead of these round trips?

Reply:
Hello. I  can fix property and resubmit pull request. What you can recommend?

Reply:
@ConstantinOr Can you please also add some tests to show that the issue is fixed?

Reply:
Ok. No problem.